src/locale-proxy.ts

```diff
@@ -0,0 +1,98 @@
+import type { LocaleDefinition } from './definitions';
+import { FakerError } from './errors/faker-error';
+
+/**
+ * A proxy for LocaleDefinitions that marks all properties as required and throws an error when an entry is accessed that is not defined.
+ */
+export type LocaleAccess = Readonly<{
+  [key in keyof LocaleDefinition]-?: Readonly<
+    Required<NonNullable<LocaleDefinition[key]>>
+  >;
+}>;
+
+/**
+ * Creates a proxy for LocaleDefinition that throws an error when a property is accessed that is not defined.
```

Comment:
```suggestion
 * Creates a proxy for LocaleDefinition that throws an error if an undefined property is accessed.
```

This suggestion minimizes the doubled "that"